### ISSUE TITLE
Unable for a player to die if not killed by another player

### ISSUE THREAD

**Issue by stefaniamak** [Terasology/LightAndShadow#145]
## Issue

When the player reaches 0 HP alone without being killed by another client (e.g. when they fall from the high ground), they do not die but continue playing normally.

## Proposed solution for now

Have the player's `death` counter increase, without increasing their `kills` counter.

![ezgif-1-60f4ca8066b5](https://user-images.githubusercontent.com/48293545/90430229-c23f4580-e0cf-11ea-9482-0405a16d9fc0.gif)

**Comment by ahv15** [comment]
Is anybody working on this issue, if not could I give this a try?

**Comment by Cervator** [comment]
Hey @ahv15 👋 Nope not aware of anybody working on this, feel free!

**Comment by ahv15** [comment]
hey @Cervator  was hoping to work on some other issue in the mean time, I was actually not able to see any sort of a kill counter or a death counter after I had started the game , am I supposed to enable anything else in singleplayer so that I can see some sort of a kill or death counter
My screen is something like this :

![Screenshot (355)](https://user-images.githubusercontent.com/52852877/91640193-4fbc5700-ea39-11ea-866a-d2f94d5337e4.png)

**Comment by ahv15** [comment]
not too sure how the gameplay is here or what exactly should be updated.

**Comment by Cervator** [comment]
Looks like you got an initial answer on Discord (wise choice to check there!), to add to that a little: the death counter is what you see if you get to where the game ends (after 5 points). I tend to just use `hspeed` in the game console then dash back and forth a few times.

With this issue resolved the player should be able to die normally and respawn, and for each time that happens their death counter visible on the game over screen should be one higher :-)

**Comment by ahv15** [comment]
Ohhk, I was wondering whether I would have to trudge across the map several times just to see the death counter once. Will make use of this handy hack then.

**Comment by Cervator** [comment]
Yep I can usually do it in 20 seconds or so. There's also a variable holding the number of rounds to win a game, you can drop that from 5 to 1 to make it even faster :-)

**Comment by stefaniamak** [comment]
@ahv15 hi 👋 thanks for working on the issue! tips to end the game quicker: 
- go to `LAUtils.java` and change `GOAL_SCORE` variable from `5` to `1`, so you need to score 1 point to end the game (already mentioned by Cervator, I just added how to do it)
- open the developer tab and gain the enemy team's by writting `give` and `redflag` or `blackflag` (it will give you 16 flags each time, but you have to do that for each point or place the rest of the flags - before you score - to the ground, and just grab them from there

Edit:
Also! When you get killed in-game, you re-appear on top of your team's platform!

**Comment by stefaniamak** [comment]
Actually having a K/D count appear within the game might be quite a nice addition 🤔 It really did slip my mind when working on the elements. Indeed it comes to a surprise when you see that there was a counter for that at the end of the game.

**Comment by ahv15** [comment]
@stefaniamak Thanks for the tips, actually few of the issues in this module seem to be related to the game over screen, hopefully if I gain a better understanding of how everything works I should be able to fix 2-3 issues in one go.

**Comment by ahv15** [comment]
Just wanted to ask if the player gets killed does he normally get teleported back to the base? I got the number of deaths to be updated but the player doesnt seem to get teleported.

**Comment by ahv15** [comment]
So is this a problem with the teleportation or does teleportation normally not happen?

**Comment by ahv15** [comment]
@stefaniamak Everything else works fine, as all the items are being dropped and the health is being restored and the death counter is also being updated but the teleportation part does not work.

**Comment by stefaniamak** [comment]
@ahv15 yes in multiplayer you get teleported to the base right away! It actually happens instantly (without a warning/notification like "you died!"), which is kind of odd tbh

**Comment by stefaniamak** [comment]
You can also test in multiplayer by running multiple builds at the same time :) start a local game and then join with the other builds!

*edit: at least that's how it is with IntelliJ, but I am pretty sure a similar thing exists in other softwares as well!

<img width="134" alt="Screenshot_17" src="https://user-images.githubusercontent.com/48293545/91875887-c1d4ac00-ec84-11ea-8650-9db1f65d580f.png">

**Comment by ahv15** [comment]
So does the teleportation part work only in multiplayer or is it supposed to work out even in singleplayer as I had tested it out in singleplayer?

**Comment by ahv15** [comment]
Maybe thats why i wasnt being teleported.

**Comment by ahv15** [comment]
I will give the multiple builds thing a try.

**Comment by stefaniamak** [comment]
I believe it is supposed to work on single-player too, as you can die alone at multy-player too 👍

**Comment by ahv15** [comment]
The doubt arised as I saw no problem with the rest of the code , will try checking it out then.

**Comment by skaldarnar** [comment]
Nice to see such activity on this :blush: @ahv15 @stefaniamak where are we with this issue right now?

**Comment by ahv15** [comment]
@skaldarnar I haven't been able to do much for the past few weeks as my laptop is being repaired, but I am hoping to take a look at this and a few other things once it comes back.

**Comment by skaldarnar** [comment]
Oh noes, hardware failures are always bad 😕 good luck with the repair!
